### PR TITLE
add options override

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: check/filesystem.lua [hunk 1]
```diff
@@ -136,4 +144,18 @@ function FileSystemCheck:run(callback)
   callback(checkResult)
 end
 
+function GetOptionsStringForFs(mountPoint)
+  local procMountsFilename = env.get('TEST_PROC_MOUNTS') or '/proc/mounts'
+  local procMountsData, err = lfs.readFileSync(procMountsFilename)
+  if err then return nil end
+  for line in procMountsData:gmatch"(.-)\n" do
```
Comment (creationix) [inline check/filesystem.lua]: Neat, I didn't know about the `-` operator.  Looks like a non-greedy version of `*`.

File: tests/test-check-fs.lua [hunk 2]
```diff
@@ -59,7 +59,7 @@ require('tap')(function(test)
       assert(metrics['used']['t'] == 'int64')
       assert(metrics['avail']['t'] == 'int64')
       p(metrics['options'])
-      assert(metrics['options']['v']:find('hello_world'))
+      assert(metrics['options']['v']:find('hello_world') > 0)
```
Comment (creationix) [inline tests/test-check-fs.lua]: Why greater than zero?  `0` is truthy in lua and find returns `nil` if it's not found.

**Comment by creationix** [comment]
+1

File: tests/test-check-fs.lua [hunk 3]
```diff
@@ -37,6 +39,46 @@ require('tap')(function(test)
     check:run(expect(onResult))
   end)
 
+  test('check filesystem valid proc mounts override', function(expect)
+    local fs_target = '/'
+    if los.type() == "win32" then return end
```
Comment (rjemanuele) [inline tests/test-check-fs.lua]: if you have this for you need the 'if' later on?

File: tests/test-check-fs.lua [hunk 4]
```diff
@@ -37,6 +39,46 @@ require('tap')(function(test)
...
+      p(metrics['options'])
+      assert(metrics['options']['v']:find('hello_world') > 0)
+  
+      assert(tonumber(metrics['free']['v']) <= tonumber(metrics['total']['v']))
+      assert(tonumber(metrics['free']['v']) + tonumber(metrics['used']['v']) ==
+                   tonumber(metrics['total']['v']))
+  
+      -- These metrics are unavailalbe on Win32, see:
+      -- http://www.hyperic.com/support/docs/sigar/org/hyperic/sigar/FileSystemUsage.html#getFiles()
+      if los.type() ~= "win32" then
```
Comment (rjemanuele) [inline tests/test-check-fs.lua]: see above note

File: tests/test-check-fs.lua (hunk 3 above)
Comment (rphillips) [inline tests/test-check-fs.lua]: good point. fixed in 5759aa8

**Comment by rjemanuele** [comment]
+1

**Pull request merged**